addons/binding/org.openhab.binding.nuki/README.md

```diff
@@ -0,0 +1,116 @@
+
+## Supported Things
+
+Nuki Smart Lock(s) which is/are paired via Bluetooth with the Nuki Bridge.
+
+## Discovery
+
+There is no automatic discovery yet implemented.
+
+## Configuration
+
+It is absolutely recommended to configure static IP addresses for both, the openHAB2 server and the Nuki Bridge!
+
+### Nuki Bridge Configuration
+
+**Please note:** At the moment you still need to manually configure a callback URL on the Nuki Bridge which points to your openHAB2 server.  
+See [Bridge HTTP-API](https://nuki.io/en/api/), Section */callback* and */callback/add*.  
+
+For example, if your openHAB2 server's IP address is 192.168.0.100 and your Nuki Bridge's IP address is 192.168.0.50, add a callback URI to **http://192.168.0.100/nuki/bcb** on the Nuki Bridge via the [Bridge HTTP-API](https://nuki.io/en/api/) like this:
+```
+http://192.168.0.50:8080/callback/add?token=1a2b3c4d5e&url=http%3A%2F%2F192.168.0.100%3A8080%2Fnuki%2Fbcb
+```
+The callback URL has to be /nuki/bcb
+
+
+### Bridge Configuration (in openHAB2)
+
+There are several configuration parameters for a bridge:
+- **IP Address** (required)  
+The static IP address of the Nuki Bridge.
+
+- **Port** (required)  
+The Port on which the Nuki Bridge REST API runs.
+
+- **API Token** (required)  
+The API Token configured via the Nuki App when enabling the API/IFTTT.
+
+The syntax for a bridge is:
+```
+Bridge nuki:bridge:<UNIQUENAME> [ <CONFIGURATION_PARAMETERS> ]
+```
+
+### Thing Configuration
+There is just one configuration parameter for a thing:  
+- **Nuki ID** (required)  
+Nuki ID
+The ID of the Nuki Smart Lock.
+
+The syntax for a thing is:
+```
+Thing smartLock <UNIQUENAME> "<DISPLAYNAME>" @ "<LOCATION>" [ <CONFIGURATION_PARAMETER> ]
+```
+
+### Manual configuration of Bridge and Thing (.things file)
+To manually configure a Nuki Bridge and a Nuki Smart Lock in your .items file you can do the following:
+```
+Bridge nuki:bridge:NB1 [ ip="192.168.0.50", port=8080, apiToken="1a2b3c4d5e" ] {
+    Thing smartLock SL1 [ nukiId="123456789" ]
+}
+
+```
+
+### Items Configuration (.items file)
+In the .items file you can define e.g. a Switch.  
+Following the Manual configuration example from above you can do the following:
+```
+Switch Frontdoor_OpenClose      "Frontdoor (Open / Close)"      <switch> { channel="nuki:smartLock:NB1:SL1:smartLockOpenClose" }
+Switch Frontdoor_UnlatchClose   "Frontdoor (Unlatch / Close)"   <switch> { channel="nuki:smartLock:NB1:SL1:smartLockUnlatchClose" }
+```
+
+### Sitemap Configuration (.sitemap file)
+In the .sitemap file you can define e.g. a Switch.  
+Following the Manual configuration example from above you can do the following:
+```
+sitemap nuki label="Nuki Smart Lock" {
+    Switch item=Frontdoor_OpenClose    label="Frontdoor (Open / Close)"
+    Switch item=Frontdoor_UnlatchClose label="Frontdoor (Unlatch / Close)"
+}
+```
+
+
+## Troubleshooting, Debugging and Tracing
```

Comment:
Well, then let me say: "there must not be anything binding specific about logging". The very same mechanisms should apply to all bindings. And all this is documented at http://docs.openhab.org/administration/logging.html - if you feel that something relevant is missing there, feel free to do a PR against that page, so that the info is available in general.